File: components/content_capture/android/java/src/org/chromium/components/content_capture/ContentCaptureData.java
// Copyright 2019 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.content_capture;

import android.graphics.Rect;

import androidx.annotation.VisibleForTesting;

import org.jni_zero.CalledByNative;

/** The class is Java's representative of components/content_capture/common/content_capture_data.h */
public class ContentCaptureData extends ContentCaptureDataBase {
    private String mValue;

    @CalledByNative
    @VisibleForTesting
    public static ContentCaptureData createContentCaptureData(
            Object parent, long id, String value, int x, int y, int width, int height) {
        ContentCaptureData data = new ContentCaptureData(id, value, x, y, width, height);
        if (parent != null) {
            ((ContentCaptureDataBase) parent).addChild(data);
        }
        return data;
    }

    private ContentCaptureData(long id, String value, int x, int y, int width, int height) {
        super(id, new Rect(x, y, x + width, y + height));
        mValue = value;
    }

    public String getValue() {
        return mValue;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder(super.toString());
        sb.append(" value:");
        sb.append(mValue);
        return sb.toString();
    }

    @Override
    public String getText() {
        return getValue();
    }
}

File: components/content_capture/common/content_capture_data.cc
// Copyright 2019 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/content_capture/common/content_capture_data.h"

namespace content_capture {

ContentCaptureData::ContentCaptureData() = default;

ContentCaptureData::ContentCaptureData(const ContentCaptureData& data) =
    default;

ContentCaptureData::~ContentCaptureData() = default;

bool ContentCaptureData::operator==(const ContentCaptureData& other) const {
  return id == other.id && value == other.value && bounds == other.bounds &&
         children == other.children;
}

}  // namespace content_capture
